### ISSUE TITLE
[Find/Replace]: Classic Dialog doesn't handle RegEx-Errors well

### ISSUE THREAD

**Issue by fedejeanne** [vi-eclipse/Eclipse-Platform#124]
(_Taken from https://github.com/eclipse-platform/eclipse.platform.ui/issues/1204_)

# The problems
- The validation of the regular expression in the "old" search dialog (`Ctrl` + `f`  in Windows) is not displayed correctly.
![grafik](https://github.com/eclipse-platform/eclipse.platform.ui/assets/16443184/dfa4af53-767e-4f30-b9be-026859c24304)

# Goal
- The result of the validation should be shown in a decoration (`ControlDecoration`), with an error icon instead of the red text at the bottom.
- The text of the decoration should contain the details of the validation result _e.g._ "_Unclosed character near..._"
- The same should be used for the _Search_ dialog (`org.eclipse.search.internal.ui.SearchDialog`) and for the new search overlay.
- The decoration should not interfere with the existing decoration that is used to hint the _content assist_ (all 3 places have it) so the new validation decoration should be placed using `SWT.CENTER` as a style-bit.
- If possible, the new decoration should be used in all 3 places to avoid code duplication

# Non-goals
- View to validate regular expressions mentioned in [comment-1759158486](https://github.com/eclipse-platform/eclipse.platform.ui/issues/1204#issuecomment-1759158486)

**Comment by fedejeanne** [comment]
Maybe @jakub-suliga and/or @jannisCode would like to chat about it with @Wittmaxi and support him/take over?

**Comment by Wittmaxi** [comment]
@fedejeanne the mentionned issue is a good first issue for new contributors.
The main issue here is in finding an expected solution - we could not yet agree on an expected behavior (it might be helpful to create a few different prototypes ?)

Moreover, the Find/Replace overlay does not offer any feedback on RegEx-Errors (thus slightly losing functionality). Maybe we can find a solution for the Overlay as well?

I am not currently planning on fixing the above issue, I believe it is an excellent issue e.g. for @jannisCode so he can quickly see first results. Feel free to reach out :)

**Comment by fedejeanne** [comment]
Thank you for the details, @Wittmaxi! Then I'll refine this issue and set some goals and proposals in the coming days.

> Moreover, the Find/Replace overlay does not offer any feedback on RegEx-Errors (thus slightly losing functionality). Maybe we can find a solution for the Overlay as well?

I'll keep that in mind 👍

**Comment by fedejeanne** [comment]
@jannisCode I changed the description of this issue to reflect what we discussed. I left it in this internal issue because I merely accepted the proposals that were made in the original issue (https://github.com/eclipse-platform/eclipse.platform.ui/issues/1204) so the PR can simply reference that issue.

@Wittmaxi I left out the part about the view to evaluate the expressions (see _Non-goals_). That one can be done in a separate PR/Issue.
